flow/connectors/core.go

```diff
@@ -264,13 +265,19 @@ func LoadPeerType(ctx context.Context, catalogPool *pgxpool.Pool, peerName strin
 
 func LoadPeer(ctx context.Context, catalogPool *pgxpool.Pool, peerName string) (*protos.Peer, error) {
 	row := catalogPool.QueryRow(ctx, `
-		SELECT type, options
+		SELECT type, options, enc_key_id
 		FROM peers
 		WHERE name = $1`, peerName)
 
 	peer := &protos.Peer{Name: peerName}
-	var peerOptions []byte
-	if err := row.Scan(&peer.Type, &peerOptions); err != nil {
+	var encKeyID string
+	var encPeerOptions []byte
+	if err := row.Scan(&peer.Type, &encPeerOptions, &encKeyID); err != nil {
+		return nil, fmt.Errorf("failed to load peer: %w", err)
+	}
+
+	peerOptions, err := decryptPeerOptions(encKeyID, encPeerOptions)
+	if err != nil {
 		return nil, fmt.Errorf("failed to load peer: %w", err)
```

Comment:
error should be swallowed at some point

Reply:
?

Reply:
talked with Kevin, he's suggesting crypto related error messages shouldn't display to user, can review crypto error messages for leaks later